mastering-plone/behaviors_2.rst

```diff
@@ -172,40 +174,50 @@ The interfaces need to be written, in our case into a file :file:`interfaces.py`
             Clear the votes. Should only be called by admins
             """
 
-    alsoProvides(IVoting, IFormFieldProvider)
-
 
 .. only:: not presentation
 
     This is a lot of code. The IVotableLayer we will need later for viewlets and browser views. Let's add it right here.
     The IVotable interface is the simple marker interface. It will only be used to bind browser views and viewlets to contenttypes that provide our behavior, so no code needed.
 
-    The IVoting class is more complex, as you can see. While IVoting is just an interface, we use :samp:`plone.supermodel.model.Schema` for advanced dexterity features.
-    Zope.schema provides no means for hiding fields. The directives :samp:`form.omitted` from :samp:`plone.autoform` allow us to annotate this additional information so that the autoform renderers for forms can use the additional information.
+    The IVoting class is more complex, as you can see.
+
+    The :samp:`@provider` decorator above the class ensures that the schema fields are known to other packages.
+    Whenever some code wants all schemas from an object, it receives the schema defined directly on the object and the additional schemata.
+    Additional schemata are compiled by looking for behaviors and whether they provide the :samp:`IFormFieldProvider` functionality.
+    Only then the fields are used as form fields.
+
+    While IVoting is just an interface, we use :samp:`plone.supermodel.model.Schema` for advanced dexterity features.
+    Zope.schema provides no means for hiding fields.
 
+    The directives :samp:`form.omitted` from :samp:`plone.autoform` allow us to annotate this additional information so that the autoform renderers for forms can use the additional information.
     We make this omit conditional. If we run Plone in debug mode, we will be able to see the internal data in the edit form.
 
-    We create minimal schema fields for our internal data structures. For a small test, I removed the form omitted directives and opened the edit view of a talk that uses the behavior. After seeing the ugliness, I decided that I should provide at least  minimum of information. Titles and required are purely optional, but very helpful if the fields won't be omitted, something that can be helpful when debugging the behavior.
+    We create minimal schema fields for our internal data structures.
+    For a small test, I removed the form omitted directives and opened the edit view of a talk that uses the behavior. After seeing the ugliness, I decided that I should provide at least  minimum of information.
```

Comment:
Place each sentence on its own line.

Reply:
Unfortunately, in the training docs the existing style is f.... has room for improvements :)